### ISSUE TITLE
[Notification][WNS] have a way to register into WNS throught a proxy (get channel URI)

### ISSUE THREAD

**Issue by ice02** [microsoft/WindowsAppSDK#1526]
# Proposal: Give a way to use default proxy or to give proxy information to register into WNS (to generate channel URI)

## Summary
Today, to register into WNS (to get channel URI), a direct connection seems to be mandatory. 
With a valid channel, all is working fine behind proxy
It could be great to let channel registration using a proxy to be able to use WNS notification behind corporate proxy

## Rationale
* First, this kind of feature is mandatory to let large company use WNS
* Second, it's more secure to let proxified connection manage it rather than direct connections

**Comment by hulumane** [comment]
@nicuparente We have proxy support in our backlog correct?

**Comment by hulumane** [comment]
+ @andrewleader Can we add this to Product Board for tracking purposes? Or is there some other avenue?

**Comment by andrewleader** [comment]
@hulumane yeah we can add it! I'll work with you and Nicu offline on what you want to put up there!

**Comment by hulumane** [comment]
It is interesting to note here that the feature isn't specific to the app specific channel registration per se. We just need to support WNS connections for all apps over a proxy in corp environments. Today the proxies don't understand custom protocol and block our traffic without an endpoint exemption. So general guidance has been to add an exemption to the proxy environment for WNS endpoints to allow the traffic through. @nicuparente  I recall there being a documentation link for this exemption stuff? May be worth following up on this option in the interim.

**Comment by nicuparente** [comment]
Correct - the current guidance is to allow list the traffic. Here is the [documentation](https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/firewall-allowlist-config) and I believe @ice02 came across this doc then followed our guidance there to file this as feedback in the Windows App SDK repo.
